d2/src/test/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancerTest.java

```diff
@@ -439,117 +439,81 @@ public void testClusterInfoProviderUnregisterClusterListener()
   }
 
   @Test
-  public void testListenToServiceAndClusterTimeout() throws Exception
+  public void testListenToServiceAndClusterTimeout()
   {
     MockStore<ServiceProperties> serviceRegistry = new MockStore<>();
     MockStore<ClusterProperties> clusterRegistry = new MockStore<>();
     MockStore<UriProperties> uriRegistry = new MockStore<>();
-    LoadBalancerTestState state = spy(new LoadBalancerTestState());
-    state.listenToService = false;
-    SimpleLoadBalancer loadBalancer = spy(setupLoadBalancer(state, serviceRegistry, clusterRegistry, uriRegistry));
+    SimpleLoadBalancerState state =
+        spy(new SimpleLoadBalancerState(new SynchronousExecutorService(), uriRegistry, clusterRegistry, serviceRegistry,
+                                        new HashMap<>(), new HashMap<>()));
+    doAnswer(invocation ->
+             {
+               Thread.sleep(100);
```

Comment:
Looks like this will block the test thread? Can we try to avoid that and if that's unavoidable, can we set a lower sleep duration, like 10ms?

Reply:
I updated the code to `async` mock. Now it's ok to set 10ms sleep duration